lib/storage/src/content_manager/toc/point_ops.rs

```diff
@@ -31,13 +32,17 @@ impl TableOfContent {
     pub async fn recommend(
         &self,
         collection_name: &str,
-        request: RecommendRequestInternal,
+        mut request: RecommendRequestInternal,
         read_consistency: Option<ReadConsistency>,
         shard_selector: ShardSelectorInternal,
-        _claims: Option<Claims>,
+        claims: Option<Claims>,
         timeout: Option<Duration>,
     ) -> Result<Vec<ScoredPoint>, StorageError> {
-        // TODO(RBAC): handle claims
+        if let Some(claims) = claims.as_ref() {
+            check_collection_name(claims, collection_name)?;
+            check_points_op(claims, &mut request)?;
+        }
```

Comment:
Let's destructure `Claims` instead, so that we get nice compiler errors when adding/changing fields.

Reply:
Something like this

```suggestion
        if let Some(claims) = claims {
            let Claims {
                exp: _,
                w: _,
                collections,
                payload,
            } = claims;

            check_collection_name(collections, collection_name)?;
            check_points_op(payload, &mut request)?;
        }
```

Reply:
Updated.

But OTOH, now we have `check_collection_name` accepting `&Option<Vec<String>>` instead of `&Claims`, and I think it's less expressive and a bit more error-prone. So I'm not sure that this is a strict improvement.